internal/controller/mantlebackupconfig_controller.go

```diff
@@ -0,0 +1,290 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	mantlev1 "github.com/cybozu-go/mantle/api/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	MantleBackupConfigFinalizerName     = "mantle.cybozu.io/mbc-finalizer"
+	MantleBackupConfigCronJobNamePrefix = "mbc-"
+)
+
+// MantleBackupConfigReconciler reconciles a MantleBackupConfig object
+type MantleBackupConfigReconciler struct {
+	Client               client.Client
+	Scheme               *runtime.Scheme
+	managedCephClusterID string
+	expireOffset         string
+}
+
+func NewMantleBackupConfigReconciler(cli client.Client, scheme *runtime.Scheme, managedCephClusterID string, expireOffset string) *MantleBackupConfigReconciler {
+	return &MantleBackupConfigReconciler{cli, scheme, managedCephClusterID, expireOffset}
+}
+
+//+kubebuilder:rbac:groups=mantle.cybozu.io,resources=mantlebackupconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=mantle.cybozu.io,resources=mantlebackupconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=mantle.cybozu.io,resources=mantlebackupconfigs/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups=batch,resources=cronjobs,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the MantleBackupConfig object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.3/pkg/reconcile
+func (r *MantleBackupConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Get the running pod info
+	runningPod, err := getRunningPod(ctx, r.Client)
```

Comment:
```suggestion
	cronJobInfo, err := getCronJobInfo(ctx, r.Client)
```

How about remaning the function to exlain the role of this function clearly? In addition, it's better to define the dedicated struct for cronJobInfo because this informatino is not about the controller's pod itself. The new struct would be like this.

```
struct cronJobInfo {
  Namespace
  ServiceAccountName
  Image
}
```
  
```

Reply:
I mostly agree with you. We should probably define two separate functions: `getCronJobInfo` and `getRunningPod`. Then, `getCronJobInfo` can call `getRunningPod`. This way, `getRunningPod` can be reused for other purposes, even though it's currently only used for creating CronJobs.